specification/app/resource-manager/Microsoft.App/stable/2022-05-01/CommonDefinitions.json

```diff
@@ -468,6 +468,16 @@
           "description": "User friendly suffix that is appended to the revision name",
           "type": "string"
         },
+        "initContainers": {
```

Comment:
You have to update or create new resource for init container. 
Refer `specification\app\resource-manager\Microsoft.App\preview\2022-06-01-preview\ContainerApps.json`

Reply:
Is this not creating the resource for initContainers? In ContainerApps.json, the template is defined by linking to the template definition here in CommonDefinitions that I've changed

            "template": {
              "$ref": "./CommonDefinitions.json#/definitions/Template",